packages/playwright/src/runner/dispatcher.ts

```diff
@@ -412,11 +412,18 @@ class JobDispatcher {
     // - there are no remaining
     // - we are here not because something failed
     // - no unrecoverable worker error
-    if (!this._remainingByTestId.size && !this._failedTests.size && !params.fatalErrors.length && !params.skipTestsDueToSetupFailure.length && !params.fatalUnknownTestIds && !params.unexpectedExitError) {
+    if (!this._remainingByTestId.size && !this._failedTests.size && !params.fatalErrors.length && !params.skipTestsDueToSetupFailure.length && !params.fatalUnknownTestIds && !params.missingProjectById && !params.unexpectedExitError) {
       this._finished({ didFail: false });
       return;
     }
 
+    if (params.missingProjectById) {
+      this._failureTracker.onWorkerError();
+      this._reporter.onError?.({ message: `Project with name ${params.missingProjectById} not found. Make sure project name does not change.` });
```

Comment:
- This place should already know the project through `this._job.projectId`.
- We should display the project name to the user, not the project id.